lib/router.js

```diff
@@ -421,8 +421,9 @@ Router.prototype._handlePaths = function(rootNode, spec, scope) {
 
     // Handle paths
     // Sequence the build process with `.each` to avoid race conditions
-    // while building the tree.
-    return P.each(Object.keys(paths), function(pathPattern) {
+    // while building the tree. Also sort(), so that fixed path segments sort
+    // before patterns (by virtue of `[a-zA-Z/] < '{'`).
+    return P.each(Object.keys(paths).sort(), function(pathPattern) {
```

Comment:
Just looked into this, but realized that it is actually not that easy. The spec paths are assigned / merged in path processing order, and we want to process those after sorting. So, I don't see an easy way to preserve the original order if we want to make sure that paths are processed deterministically.